appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -1043,7 +1043,6 @@ public PayaraMicroRuntime bootStrap() throws BootstrapException {
 
             // boot the server
             preBootCommands.executeCommands(gf.getCommandRunner());
-            HazelcastCore.setThreadLocalDisabled(true);
             try {
```

Comment:
If this solves the issue and we agree, that this is how hazelcast bootstrap should work (pre-boot, rather than post boot), then entire code for handling "threadlocaldisabled" should be removed as well, as it has no other usage.